File: FormalBench-data/diverse/rule_21/TriangleArea.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;

class TriangleArea {

	public static int triangleArea(int r) {
		int area = 0;
		if (r < 0) {
			return -1;
		} else if (0 == r) {
			area = 0;
		} else {
			area = (int) Math.sqrt(3) * r * r;
		}
		return area;
	}
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/TriangleArea.c
#include "java_arrays.h"
#include <math.h>
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left * right);
*/
static int32_t java_mul(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left * (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (r < 0 ? -1 : (int32_t)((integer)r * r));
*/
int32_t triangleArea(int32_t r) {
    int32_t area = 0;
    if (r < 0) {
        return -1;
    } else if (r == 0) {
        area = 0;
    } else {
        int32_t coefficient = (int32_t)sqrt(3.0);
        area = java_mul(java_mul(coefficient, r), r);
    }
    return area;
}
